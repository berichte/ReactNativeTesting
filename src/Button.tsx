import React from 'react';
import {Text, TouchableOpacity} from 'react-native';

type ButtonProps = {
  title: string;
  onPress: () => unknown;
};
export function Button({title, onPress}: ButtonProps) {
  return (
    <TouchableOpacity testID="my-button" onPress={onPress}>
      <Text>{title}</Text>
    </TouchableOpacity>
  );
}
