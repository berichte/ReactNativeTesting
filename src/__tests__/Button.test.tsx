import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';

import {Button} from '../Button';

describe('Button', () => {
  it('should render the title "A Button for testing"', () => {
    const {toJSON} = render(
      <Button title="A Button for testing" onPress={() => {}} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('should render the title "TESTING"', () => {
    const {toJSON} = render(<Button title="TESTING" onPress={() => {}} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('should call onPress when pressed', () => {
    const onPress = jest.fn();
    render(<Button title="A Button for testing" onPress={onPress} />);
    fireEvent(screen.getByTestId('my-button'), 'press');
    expect(onPress).toHaveBeenCalled();
  });
});
