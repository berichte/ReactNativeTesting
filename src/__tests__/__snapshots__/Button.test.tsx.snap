// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the title "A Button for testing" 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="my-button"
>
  <Text>
    A Button for testing
  </Text>
</View>
`;

exports[`Button should render the title "TESTING" 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="my-button"
>
  <Text>
    TESTING
  </Text>
</View>
`;
